core/src/test/resources/newlines/alwaysBeforeTopLevelStatements.stat

```diff
@@ -144,4 +158,117 @@ object A {
 
   @Singleton
   class B @Inject()(val x: Int)
+
+  @js.native
+  class C() {
+    def c = 5
+  }
+}
+<<< Double annotation
+object a {
+  @annot
+  @bannot
+  class A
+}
+>>>
+object a {
+
+  @annot
+  @bannot
+  class A
+}
+<<< Don't add NL before single-line defs
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+>>>
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+
+<<< Add NL before multi-line statements
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+  def foo = {
+    println("foo")
+  }
+  def bar = {
+    println("bar")
+  }
+}
+>>>
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+<<< nested defs (up for debate)
+object a {
+  def foo = {
+    def b = {
+      3
+    }
+    def c = {
+      4
+    }
+  }
+}
+>>>
+object a {
+
+  def foo = {
+
+    def b = {
```

Comment:
I think the question is if the first definition inside (anything) should have a newline like:
```
object A {
  object B {

  }
}
```

I think for anything with lots of pameters it would look better to have a newline before the childs start:
```
class A (val a: Int. 
              val a: Int. 
              val a: Int. 
              val a: Int. 
              val a: Int
            ) {

     class B (val a: Int. 
              val a: Int. 
              val a: Int. 
              val a: Int. 
              val a: Int
            ) 
}
```

so maybe just add it generally?